rust-runtime/aws-smithy-async/src/rt/sleep.rs

```diff
@@ -39,15 +39,44 @@ where
     }
 }
 
+/// Wrapper type for sharable `AsyncSleep`
+#[derive(Clone, Debug)]
+pub struct SharedAsyncSleep(Arc<dyn AsyncSleep>);
```

Comment:
To prevent tuple struct construction:
```suggestion
#[non_exhaustive]
pub struct SharedAsyncSleep(Arc<dyn AsyncSleep>);
```

Reply:
Tuple struct construction is not allowed even without `#[non_exhaustive]`, i.e. neither compiles
```
let sleep = SharedAsyncSleep {0: Arc::new( /* type implementing AsyncSleep */ )};
```
```
let sleep = SharedAsyncSleep(Arc::new( /* type implementing AsyncSleep */ ));
```